libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1671,16 +1671,19 @@ void ExpressionCompiler::appendExternalFunctionCall(
 		utils().storeFreeMemoryPointer();
 	}
 
-	// Touch the end of the output area so that we do not pay for memory resize during the call
-	// (which we would have to subtract from the gas left)
-	// We could also just use MLOAD; POP right before the gas calculation, but the optimizer
-	// would remove that, so we use MSTORE here.
-	if (!_functionType.gasSet() && retSize > 0)
+	if (!m_context.evmVersion().canOverchargeGasForCall())
```

Comment:
Actually this code piece may still be useful if there is out of gas condition, it stops at memory expansion and not at the time of the return of the call (it is an edge case though).

Also not sure this doesn't depend on `hasReturndata` the option?

Reply:
I'm not 100% sure I understand. The problem is that the `CALL` opcode performs memory expansion for the return area. If I remember correctly, the `CALL` opcode first performs memory expansion to accomodate both the argument and the return data memory area. After that, pre-tangerine-whistle VMs would check if there is still enough gas left to send the specified amount with the call. Post-tangerine-whistle VMs would not fail if this check is negative, but just send as much gas as is still left.

"it stops .. not at the time of the return of the call" - as far as I know, this can never happen, since the memory is expanded before the call is performed. Moving this to after the call was one of the early suggestions as part of EIP5.

This should be unrelated to `hasReturndata`.

Reply:
Had to check this again also discussed with @chfast. Apparently memory expansion happens before the call and the remainder gas is taken as you've said. My comment is invalid then.

Also not including this code should make no difference in behaviour but should save gas by not doing it manually.